### DISCUSSION TITLE
Guidance on all-caps legibility?

### DISCUSSION THREAD

**Discussion by tkeenoy** [Myndex/SAPC-APCA#19] [Q&A] [answered]
Many common UI patterns adopt an all caps styling for some interface elements, notably buttons. Typically in old shcool typographic standards, this effectively puts the x height of the resulting text string roughly equivalent to the cap height of the font size selected. It's not good for anything more than 1-2 words, at which point it starts to interfere with readability (as opposed to legibility).

Is this factored into any of the research? Will it be factored into the 3.0 standard?

Issues I see:
Typically, capital letters are very slightly heavier in weight than lower case letters. However, they're not a full weight step (e.g. 400 –> 500) up, so a blanket rule of "treat it as one step up" wouldn't necessarily work. Here's a test I did using Helvetica Neue 400 and 500, comparing lower case L to capital i. 
![Screen Shot 2020-11-04 at 10 15 13 AM](https://user-images.githubusercontent.com/58079898/98152759-106e1480-1e87-11eb-97d9-7be544ad6afc.png)

Greater apparent x height won't necessarily produce better legibility based on the spatial frequency and contrast sensitivity curve. Or maybe the larger spaces inherent in capital letters do, in fact produce better legibility. This might need some empirical testing.

What does letter spacing do to the whole thing? Wider letter spacing should, hypothetically, create wider spatial frequency, improving both legibility and (to a point) readability. This is a classic typographic maneuver, especially with short titles (e.g. headlines, buttons). Letter spacing might want its own separate issue.

**Comment by Myndex** [comment, marked as answer]
Hi @tkeenoy  

Thank you for the comment, and I an glad someone is taking an interest in the minutiae of this very important subject.

On the subject of ALL-CAPS. IN GENERAL IT MAKES FLUENT READABILITY MORE DIFFICULT AS IT CAN IN THEORY  INTERFERE WITH THE VWFA (VISUAL WORD FORM AREA)'S LEARNED ABILITY TO RECOGNIZE FULL WORDS, as opposed to using lower case, which is the normal readability standard.

The standard for fluent readability, which means lexical stimulus is processed by the visual cortex and sent to the VWFA for processing as letter pairs and whole words, is based on x-height & lower-case.

Spot reading, as in "a word or two" on a button, actually has a lower contrast requirement. Capitals, with a slightly thicker weight also have a slightly lower luminance contrast requirement. 

### Relevance

Right now, we are not making exceptions for spot reading, other than for non-content such as for copyright notices, bylines, etc. Should the standard be relaxed for spot-reading such as for one or two words on a button? Maybe at some point. Right now we're trying to correct 12 years of too-low contrast on blocks of body text.


> Greater apparent x height won't necessarily produce better legibility based on the spatial frequency and contrast sensitivity curve. Or maybe the larger spaces inherent in capital letters do, in fact produce better legibility. This might need some empirical testing.

Greater x-height produces better legibility in terms of visual acuity, and there is substantial empirical testing in this area, I suggest Legge's "Psychophysics of Reading."

And legibility does not equal readability. Legibility means you can "make out what it is" letter by letter. Readability means that the VWFA can process whole words. There is an enormous difference. The Visual Contrast standard for Silver and the APCA is focused on **readability** not legibility.

> What does letter spacing do to the whole thing? Wider letter spacing should, hypothetically, create wider spatial frequency, improving both legibility and (to a point) readability. This is a classic typographic maneuver, especially with short titles (e.g. headlines, buttons). Letter spacing might want its own separate issue.

Letter and line spacing has a direct impact on visual contrast for readability. In developing standards documents, there are questions of where certain things like that or padding for instance should go, as there is a desire for a certain level of granularity in testing procedures. This is yet to be settled as this working draft moves forward.

All of the things you mention have been significant parts of the last year and a half of research and development in this area.

**Some thoughts:** On font weight, there is in fact no consistent standard for what makes a weight 200 for instance. We're working on some programatic means to determine this.

At present, the standard defines letter spacing as no less than that designed into the font. While increasing spacing might slightly lower the contrast requirement, there is not a "bonus" of an allowable decrease in luminance contrast. Just as in spot reading on a button.

When in the area of readability, we're talking about contrast above the critical contrast for best whole word recognition in the VWFA. That is far far higher than for spot reading (letter by letter) or for legibility (the pure edge of visibility). I could see, once the contrast specification for objects and non-text is settled (which is processed by a different portion of the brain than the VWFA and has different contrast and color needs), then going and looking into the needs for spot reading and if a relaxed standard for the word on a button is appropriate.

But personally I want to see readability fixed first.

I hope this answers your questions.

Andy

**Comment by tkeenoy** [comment]
This is great Andy, thanks for the detail, and thanks for the reading recommendation! As a part time typography nerd, I am following this whole process with great interest.

**Comment by Myndex** [comment]
FWIW my recent article discussing uppercase, and including references to some of the latest research is live, the link is:

## [_Hats off to ALL CAPS_](https://tangledweb.xyz/hats-off-to-all-caps-c0a43a2c30d4)
Myth-busting the misunderstandings regarding accessibility: are dyslexia fonts useful and the shift in using ALL UPPERCASE LETTERS for various text elements.


[![A man standing in the field with hats being blown off... image by DallE and A.Somers](https://user-images.githubusercontent.com/42009457/209576034-8951cd2c-9d4e-43c4-a7ff-6e544455b040.png)](https://tangledweb.xyz/hats-off-to-all-caps-c0a43a2c30d4)
